source/httpfs/http_filesystem.go

```
@@ -0,0 +1,145 @@
+		}
+
+		if !s.migrations.Append(m) {
+			return nil, fmt.Errorf("unable to parse file %v", fi.Name())
+		}
+	}
+
+	return s, nil
+}
+
+// Open is not implemented as many FileSystems cannot be opened or have
+// custom initialization.
+//
+// Common use:
+//   plainFileSystem, err := WithInstance(http.Dir(tmpDir), &Config{})
+//
+//   escEmbeddedFileSystem, err := WithInstance(FS(false), &Config{})
+func (f *HTTPFS) Open(url string) (source.Driver, error) {
+	return nil, fmt.Errorf("not implemented")
+}
+
+// Close currently does not do anything, as there is no common way to release
+// FileSystem objects.
+func (f *HTTPFS) Close() error {
+	// nothing do to here
+	return nil
+}
+
+// First is implementation of source.First(). Please see it for documentation.
+func (f *HTTPFS) First() (version uint, err error) {
+	v, ok := f.migrations.First()
+	if !ok {
+		return 0, &os.PathError{Op: "first", Path: f.path, Err: os.ErrNotExist}
+	}
+
+	return v, nil
+}
+
+// Prev is implementation of source.Prev(). Please see it for documentation.
+func (f *HTTPFS) Prev(version uint) (prevVersion uint, err error) {
+	v, ok := f.migrations.Prev(version)
+	if !ok {
+		return 0, &os.PathError{Op: fmt.Sprintf("prev for version %v", version), Path: f.path, Err: os.ErrNotExist}
+	}
+
+	return v, nil
+}
+
+// Next is implementation of source.Next(). Please see it for documentation.
+func (f *HTTPFS) Next(version uint) (nextVersion uint, err error) {
+	v, ok := f.migrations.Next(version)
+	if !ok {
+		return 0, &os.PathError{Op: fmt.Sprintf("next for version %v", version), Path: f.path, Err: os.ErrNotExist}
+	}
+
+	return v, nil
+}
+
+// ReadUp is implementation of source.ReadUp(). Please see it for documentation.
+func (f *HTTPFS) ReadUp(version uint) (r io.ReadCloser, identifier string, err error) {
+	if m, ok := f.migrations.Up(version); ok {
+		p := path.Join(f.path, m.Raw)
+		r, err := f.fs.Open(p)
+		if err != nil {
+			return nil, "", fmt.Errorf("can't open file %s: %s", p, err)
```

Don't mask the original error and instead use `go-multierror`